privacy-on-beam/pbeam/mean_test.go

```diff
@@ -536,58 +533,77 @@ func TestMeanPerKeyNoNoiseFloatValues(t *testing.T) {
 
 // Checks that MeanPerKey with partitions returns a correct answer for float input values.
 func TestMeanPerKeyWithPartitionsNoNoiseFloatValues(t *testing.T) {
-	triples := concatenateTriplesWithFloatValue(
-		makeTripleWithFloatValue(7, 0, 2),
-		makeTripleWithFloatValueStartingFromKey(7, 100, 1, 1),
-		makeTripleWithFloatValueStartingFromKey(107, 150, 1, 2))
-
-	exactCount0 :=  7.0
-	exactMean0 := 14.0/exactCount0
-	exactCount := 250.0 
-	exactMean := (100.0 + 2.0*150.0) / exactCount
-	result := []testFloat64Metric{
-		{0, exactMean0},
-		{1, exactMean},
-	}
-
-	p, s, col, want := ptest.CreateList2(triples, result)
-	col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
-
-	partitions := []int{0, 1}
-	partitionsCol := beam.CreateList(s, partitions)
+	for _, tc := range []struct {
+		lower float64
+		upper float64
+	}{
+		// Used for MinValue and MaxValue. Tests case when specified partitions are already in the data.
+		{
+			lower: 1.0, 
+			upper: 3.0,
+		},
+		{
+			lower: 0.0,
+			upper: 2.0,
+		},
+		{
+			lower: -10.0,
+			upper: 10.0,
+		},
+	} {
+		triples := concatenateTriplesWithFloatValue(
+			makeTripleWithFloatValue(7, 0, 2),
+			makeTripleWithFloatValueStartingFromKey(7, 100, 1, 1),
+			makeTripleWithFloatValueStartingFromKey(107, 150, 1, 2))
+
+		exactCount0 :=  7.0
+		exactMean0 := 14.0/exactCount0
+		exactCount := 250.0 
+		exactMean := (100.0 + 2.0*150.0) / exactCount
+		result := []testFloat64Metric{
+			{0, exactMean0},
+			{1, exactMean},
+		}
 
-	// We have ε=50, δ=0 and l0Sensitivity=1. No thresholding is done because partitions are specified.
-	// We have 2 partitions. So, to get an overall flakiness of 10⁻²³,
-	// we can have each partition fail with 10⁻²⁵ probability (k=25).
-	maxContributionsPerPartition := int64(1)
-	maxPartitionsContributed := int64(1)
-	epsilon := 100.0
-	delta := 0.0
-	lower := 1.0
-	upper := 3.0
+		p, s, col, want := ptest.CreateList2(triples, result)
+		col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
 
-	// ε is not split because partitions are specified.
-	pcol := MakePrivate(s, col, NewPrivacySpec(epsilon, delta))
-	pcol = ParDo(s, tripleWithFloatValueToKV, pcol)
-	got := MeanPerKey(s, pcol, MeanParams{
-		MaxPartitionsContributed:     maxPartitionsContributed,
-		MaxContributionsPerPartition: maxContributionsPerPartition,
-		MinValue:                     lower,
-		MaxValue:                     upper,
-		NoiseKind:                    LaplaceNoise{},
-		partitionsCol: partitionsCol,
-	})
+		partitions := []int{0, 1}
+		partitionsCol := beam.CreateList(s, partitions)
+
+		// We have ε=50, δ=0 and l0Sensitivity=1. No thresholding is done because partitions are specified.
+		// We have 2 partitions. So, to get an overall flakiness of 10⁻²³,
+		// we can have each partition fail with 10⁻²⁵ probability (k=25).
+		maxContributionsPerPartition := int64(1)
+		maxPartitionsContributed := int64(1)
+		epsilon := 100.0
```

Comment:
epsilon should be 50 because it's not split.

Reply:
Please re-check the epsilon and deltas for all the tests you've written. They are easy to miss as you can see :)